src/hotspot/share/gc/shenandoah/shenandoahHeapRegion.cpp

```diff
@@ -458,7 +464,17 @@ void ShenandoahHeapRegion::oop_fill_and_coalesce() {
       heap->card_scan()->coalesce_objects(obj_addr, fill_size);
       obj_addr = next_marked_obj;
     }
+    if (!ops_before_preempt_check--) {
```

Comment:
I think the `C++` way is to compare against 0, not coerce to bool.

Reply:
Thanks.  I've changed the condition.